File: dsa/6.hash_table/2405_Optimal_Partition_of_String.java
package hash_table;
import java.util.*;

class Solution {
    public int partitionString(String s) {
        Set<Character> hashSet = new HashSet<Character>();
        int result = 0;
        for (int i = 0; i < s.length(); i++){
            if (hashSet.contains(s.charAt(i)) == true){
                result += 1;
                hashSet.clear();
            }
            hashSet.add(s.charAt(i));
        }
        return result + 1;
    }
}

File: dsa/6.hash_table/2405_Optimal_Partition_of_String.py
import typing as t

class Solution:
    def partitionString(self, s: str) -> int:
        track_char_set: t.Set[str] = set()
        result: int = 0
        for ind in range(0, len(s), 1):
            if s[ind] in track_char_set:
                track_char_set.clear()
                result += 1
            track_char_set.add(s[ind])
        return result + 1